src/components/tag-input/tag-input.component.ts

```diff
@@ -336,10 +336,10 @@ export class TagInputComponent<T = any> implements AfterContentInit, OnChanges,
         }
 
         // Get the input field cursor location
-        const inputCursorPos = this.tagInput.nativeElement.selectionStart;
+        const inputCursorPos = this.tagInput?.nativeElement.selectionStart;

```

Comment:
Could we update the ViewChild statement to include the correct generic type. Currently typescript is inferring `inputCursorPos` as `any` rather than `number`.

Also, right now whenever there is no input `inputCursorPos` is `undefined`, however later we are assuming it is a number, e.g.:

```ts
inputCursorPos <= 0
```

This seems like it might be a bit unsafe, it might be good to supply a default value for `inputCursorPos` whenever the input element is undefined. (I'm not sure what the correct value should be, but something like this):

```ts
const inputCursorPos = this.tagInput?.nativeElement.selectionStart ?? 0;
```

Alternatively we could update `canNavigateLeft` and `canNavigateRight` to check if `inptuCursorPos` is `undefined` before using it.